pkg/storage/unified/search/bleve.go

```diff
@@ -554,6 +550,56 @@ func toBleveSearchRequest(req *resource.ResourceSearchRequest, access authz.Acce
 		searchrequest.Query = bleve.NewConjunctionQuery(queries...) // AND
 	}
 
+	// Can we remove this? Is access ever nil?
+	if access != nil {
+		auth, ok := claims.From(ctx)
+		if !ok {
+			return nil, resource.AsErrorResult(fmt.Errorf("missing claims"))
+		}
+		folderChecker, err := access.Compile(ctx, auth, authz.ListRequest{
+			Namespace: b.key.Namespace,
+			Group:     b.key.Group,
+			Resource:  "folder",
+			Verb:      utils.VerbList,
+		})
+		if err != nil {
+			return nil, resource.AsErrorResult(err)
+		}
+		dashboardChecker, err := access.Compile(ctx, auth, authz.ListRequest{
+			Namespace: b.key.Namespace,
+			Group:     b.key.Group,
+			Resource:  "dashboard",
+			Verb:      utils.VerbList,
+		})
```

Comment:
```suggestion
		checker, err := access.Compile(ctx, auth, authz.ListRequest{
			Namespace: b.key.Namespace,
			Group:     b.key.Group,
			Resource:  b.key.Resource,
			Verb:      utils.VerbList,
		})
```
The index is tied to a single resource type that is in the key -- that is the checker that will be valid for this kind

Reply:
✅

Reply:
When federated -- this will need to make sure it is looking at the correct checker.  You can still initalize them at the beginning, but each check would need to read from the correct checker.